### PR TITLE
feat(pg): Support Taiko Katla, Mode, Mode Sepolia, Polygon ZKEVM Sepolia, Linea Sepolia

### PR DESCRIPTION
**Pull request by RPate97**
## Purpose
Adds support for:
- Taiko Katla
- Mode
- Mode Sepolia
- Polygon ZKEvm Sepolia
- Linea Sepolia

We still have to add support for Polygon Amoy (the replacement for Mumbai), but I'm waiting on test net tokens. They have a process for infra providers to request bulk tokens, so I'm trying that.

I decided not to deprecate the remaining Goerli-related chains just yet; I'll do that closer to the deadline.

### CODE REVIEW

File: packages/core/test/convert.spec.ts [hunk 1]
```diff
@@ -77,6 +77,13 @@ describe('Convert EthersJS Objects', () => {
     81457: '0x9bef99e8490336c4f166fbe8864ec9a8a19267960674d3103573aa818a71df56',
     168587773:
       '0xb2919ae35bb27081349e37d523d326da9d9bd8e08b120e5cc8404b7dfe55ec15',
+    2442: '0xe271bde4a228c8bb5363bb3f9c22345612260c04c7a78eaada880eefd2deec14',
+    43: '',
```
Comment (RPate97) [inline packages/core/test/convert.spec.ts:81]: I'm waiting on tokens for Darwinia Pangolin.

File: packages/contracts/src/networks.ts [hunk 2]
```diff
@@ -1195,4 +1243,90 @@ export const SPHINX_NETWORKS: Array<SupportedNetwork> = [
...
+    actionGasLimitBuffer: false,
+    useHigherMaxGasLimit: false,
+    eip2028: true,
+    requiredEnvVariables: ['TAIKO_KATLA_RPC_URL'],
+  },
+  {
+    name: 'darwinia_pangolin',
+    displayName: 'Darwinia Pangolin Testnet',
+    chainId: BigInt(43),
+    rpcUrl: () => process.env.DARWINIA_PANGOLIN_RPC_ULR!,
```
Comment (sam-goldman) [inline packages/contracts/src/networks.ts:1268]: ULR -> URL. (There are three instances of this typo in this PR)

File: packages/contracts/src/networks.ts [hunk 3]
```diff
@@ -1195,4 +1243,90 @@ export const SPHINX_NETWORKS: Array<SupportedNetwork> = [
...
+    queryFilterBlockLimit: 2000,
+    legacyTx: false,
+    actionGasLimitBuffer: false,
+    useHigherMaxGasLimit: false,
+    eip2028: true,
+    requiredEnvVariables: ['TAIKO_KATLA_RPC_URL'],
+  },
+  {
+    name: 'darwinia_pangolin',
+    displayName: 'Darwinia Pangolin Testnet',
```
Comment (sam-goldman) [inline packages/contracts/src/networks.ts:1266]: nit: "Pangolin Testnet" seems redundant. could drop "Testnet"

**Review by sam-goldman — APPROVED** [review]
Looks good, just a couple minor comments. Thx for updating the RPC Endpoints notion

File: packages/contracts/src/networks.ts [hunk 4]
```diff
@@ -538,6 +562,30 @@ export const SPHINX_NETWORKS: Array<SupportedNetwork> = [
     useHigherMaxGasLimit: false,
     eip2028: true,
   },
+  {
+    name: 'polygon_zkevm_cardona',
+    displayName: 'Polygon zkEVM Cardona',
```
Comment (sam-goldman) [inline packages/contracts/src/networks.ts:567]: Should this be Cardona or Sepolia? The main README says Sepolia

**Review by sam-goldman — APPROVED** [review]
One other comment

**Pull request merged**